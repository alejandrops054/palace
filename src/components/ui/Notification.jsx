import React from "react";
import { Link } from "react-router-dom";

function Notification({ col = 3 }) {
  const data = [
    {
      icon: "user-check",
      message: "Pending John Does data verification",
    },
    {
      icon: "envelope",
      message: "Lorem ipsum dolor sit armet...",
    },
    {
        icon:'calendar',
      message: "new meeting schedule for...",
    },
  ];

  const ListItems = data.map((data) => (
    <div className="row" key={data.icon}>
      <div className="col mb-2">
        <i className={`mr-2 fas fa-${data.icon}`}></i>
        {data.message}
      </div>
    </div>
  ));

  return (
    <div className={`col-${col}`}>
      <h4>These need your attention</h4>
      <div className="row">{ListItems}</div>
      <div className="row">
        <div className="col align-self-center">
          <Link to="/">View All</Link>
        </div>
      </div>
    </div>
  );
}

export default Notification;
