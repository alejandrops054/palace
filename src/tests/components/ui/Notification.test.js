import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import Notification from '../../../components/ui/Notification';

describe('Pueba en <Notification/>', () => {
    test('Debe renderizar notificaciones', () => {

        const wrapper = shallow(
            <Notification/>
        );

        expect(wrapper).toMatchSnapshot();
    })

})