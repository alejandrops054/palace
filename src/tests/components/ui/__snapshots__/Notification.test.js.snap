// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pueba en <Notification/> Debe renderizar notificaciones 1`] = `
<div
  className="col-3"
>
  <h4>
    These need your attention
  </h4>
  <div
    className="row"
  >
    <div
      className="row"
      key="user-check"
    >
      <div
        className="col mb-2"
      >
        <i
          className="mr-2 fas fa-user-check"
        />
        Pending John Does data verification
      </div>
    </div>
    <div
      className="row"
      key="envelope"
    >
      <div
        className="col mb-2"
      >
        <i
          className="mr-2 fas fa-envelope"
        />
        Lorem ipsum dolor sit armet...
      </div>
    </div>
    <div
      className="row"
      key="calendar"
    >
      <div
        className="col mb-2"
      >
        <i
          className="mr-2 fas fa-calendar"
        />
        new meeting schedule for...
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col align-self-center"
    >
      <Link
        to="/"
      >
        View All
      </Link>
    </div>
  </div>
</div>
`;
